### PR TITLE
Adds information to redirect_std* inline docs

### PR DESCRIPTION
**Pull request by nishtha981**
Partially addresses issue #35959

Adds examples for redirecting std to a file to the markdown file

### CODE REVIEW

File: doc/src/manual/networking-and-streams.md [hunk 1]
```diff
@@ -184,6 +185,23 @@ julia> open("hello.txt") do f
 "HELLO AGAIN."
 ```
 
+If you want to redirect stdout to a file
+
+```# Open file for writing
+out_file = open("output.txt", "w")
+
+# Redirect stdout to file
+stdout = out_file
```
Comment (Seelengrab) [inline doc/src/manual/networking-and-streams.md:194]: This does not redirect `stdout` in code following this, that already uses the `stdout` variable internally. It's just rebinding the variable here. Redirecting `stdout` to a file is done like so:

```
redirect_stdout(open("output.txt")) do
    # your code here
end
```

**Comment by Seelengrab** [comment]
The changes here only partially address https://github.com/JuliaLang/julia/issues/35959 - the majority of the issue, about the current documentation being confusing for people not already familiar with pipes, TTY, streams, C-stdout.. is not clarified here.

File: doc/src/manual/networking-and-streams.md [hunk 2]
```diff
@@ -66,8 +67,8 @@ abcd
 "abcd"
 ```
 
-Note that depending on your terminal settings, your TTY may be line buffered and might thus require
-an additional enter before the data is sent to Julia.
+Note that depending on your terminal settings, your TTY(teletype terminal) may be line buffered and might thus require an additional enter before the data is sent to Julia.
```
Comment (stevengj) [inline doc/src/manual/networking-and-streams.md:70]:
```suggestion
Note that depending on your terminal settings, your TTY ("teletype terminal") may be line buffered and might thus require an additional enter before `stdin` data is sent to Julia.
```

File: doc/src/manual/networking-and-streams.md (hunk 1 above)
Comment (stevengj) [reply doc/src/manual/networking-and-streams.md:194]: Moreover, `redirect_stdout` also redirects standard output from things like C libraries that don't use the Julia `stdout` variable.

File: doc/src/manual/networking-and-streams.md [hunk 3]
```diff
@@ -184,6 +185,23 @@ julia> open("hello.txt") do f
...
+
+```# Open file for writing
+out_file = open("output.txt", "w")
+
+# Redirect stdout to file
+stdout = out_file
+
+# Your code here
+
+# Close file
```
Comment (vtjnash) [inline doc/src/manual/networking-and-streams.md:198]:
```suggestion
redirect_stdout(out_file) do
    # Your code here
    println("This output goes to `out_file` via the `stdout` variable.")
end

# Close file
```

**Pull request merged**